corehq/apps/accounting/tests/test_stripe_payment_method_with_api.py

```diff
@@ -0,0 +1,227 @@
+        settings.STRIPE_PRIVATE_KEY = None
+        self.assertEqual(len(self.payment_method.all_cards), 0)
+        settings.STRIPE_PRIVATE_KEY = original_key
+
+    def test_all_cards_serialized_return_the_correct_property_of_a_card(self):
+        cards = self.payment_method.all_cards_serialized(self.billing_account)
+        card = cards[0]
+        self.assertEqual(card["brand"], "Visa")
+        self.assertEqual(card["last4"], "4242")
+        # stripe might change exp date for the test card, so we don't assert absolute value
+        self.assertIn("exp_month", card)
+        self.assertIn("exp_year", card)
+        self.assertIn("token", card)
+        self.assertFalse(card["is_autopay"])
+
+    def test_get_card_return_the_correct_card_object(self):
+        # TODO: Should I rename get_card parameter to card_id?
+        card = self.payment_method.get_card(self.card.id)
+        self.assertEqual(card.id, self.card.id)
+
+    def test_get_autopay_card_when_no_autopay_card(self):
+        self.assertEqual(self.billing_account.auto_pay_user, None)
+        self.assertFalse(self.billing_account.auto_pay_enabled)
+        result = self.payment_method.get_autopay_card(self.billing_account)
+        self.assertIsNone(result)
+
+    def test_get_autopay_card_when_only_one_autopay(self):
+        self.payment_method.set_autopay(self.card, self.billing_account, None)
+        result = self.payment_method.get_autopay_card(self.billing_account)
+        self.assertEqual(result.id, self.card.id)
+
+    def test_get_autopay_card_when_one_of_many_card_is_autopay(self):
+        card2 = self.payment_method.create_card('tok_discover', self.billing_account, None, True)
+        card3 = self.payment_method.create_card('tok_amex', self.billing_account, None, True)
+        self.addCleanup(card2.delete)
+        self.addCleanup(card3.delete)
+
+        result = self.payment_method.get_autopay_card(self.billing_account)
+        self.assertIsNotNone(result.id, card2.id)
+
+        result = self.payment_method.get_autopay_card(self.billing_account)
+        self.assertIsNotNone(result.id, card3.id)
+
+    def test_remove_card_successful(self):
+        self.assertEqual(len(self.payment_method.customer.cards), 1)
+        self.payment_method.set_autopay(self.card, self.billing_account, None)
+        self.assertEqual(self.card.id, self.payment_method.get_autopay_card(self.billing_account).id)
+
+        self.payment_method.remove_card(self.card.id)
+        self.assertEqual(len(self.payment_method.customer.cards), 0)
+        self.billing_account.refresh_from_db()
+        self.assertIsNone(self.billing_account.auto_pay_user)
+        self.assertFalse(self.billing_account.auto_pay_enabled)
+
+    def test_remove_card_non_existent(self):
+        with self.assertRaises(stripe.error.InvalidRequestError):
+            self.payment_method.remove_card("non_existent_card_id")
+
+    def test_create_card_creates_card_and_sets_default(self):
```

Comment:
This test to reset default card doesn't work. But we're not using default card feature at all. Can we just remove [set_default_card()](https://github.com/dimagi/commcare-hq/blob/82c09e394da8b229085d496eb7c221d79626321f/corehq/apps/accounting/models.py#L3798) function?

Reply:
Interesting, yea we don't reference `default_card` anywhere actually charging is taking place, nor is it referenced in the UI. I think it's fine to remove this functionality. It appears stripe takes care of picking a default anyway

Reply:
Thank you. Removed in 35c916e241df7336ea6d8dce7089379dfad0e636